### PR TITLE
Simplify TrinoProcessBuilder with java executable

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

Build the java command programlly instead of a static shell script

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @hddong

**Comment by ulysses-you** [comment]
great code refactor

File: build/dist [hunk 1]
```diff
@@ -257,9 +253,8 @@ cp "$KYUUBI_HOME/externals/kyuubi-flink-sql-engine/target/kyuubi-flink-sql-engin
 cp "$KYUUBI_HOME/externals/kyuubi-spark-sql-engine/target/kyuubi-spark-sql-engine_${SCALA_VERSION}-${VERSION}.jar" "$DISTDIR/externals/engines/spark"
 
 # Copy trino engines
-cp -r "$KYUUBI_HOME/externals/kyuubi-trino-engine/bin/" "$DISTDIR/externals/engines/trino/bin/"
-chmod a+x "$DISTDIR/externals/engines/trino/bin/trino-engine.sh"
-cp "$KYUUBI_HOME/externals/kyuubi-trino-engine/target/kyuubi-trino-engine_${SCALA_VERSION}-${VERSION}.jar" "$DISTDIR/externals/engines/trino/jars"
+cp "$KYUUBI_HOME/externals/kyuubi-trino-engine/target/kyuubi-trino-engine_${SCALA_VERSION}-${VERSION}.jar" "$DISTDIR/externals/engines/trino"
+cp -r "$KYUUBI_HOME/externals/kyuubi-trino-engine/target/scala-$SCALA_VERSION/jars/" "$DISTDIR/externals/engines/trino"
```
Comment (hddong) [inline build/dist:257]: The command will put those jars under `engines/trino` directly.  Shell we make them put `trino/jars` or `trino/scala-$SCALA_VERSION/jars/`

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilder.scala [hunk 2]
```diff
@@ -17,101 +17,71 @@
...
 
-  override protected def childProcEnv: Map[String, String] = conf.getEnvs +
-    ("TRINO_ENGINE_JAR" -> mainResource.get) +
-    ("TRINO_ENGINE_DYNAMIC_ARGS" ->
-      trinoConf.map { case (k, v) => s"-D$k=$v" }.mkString(" "))
+    mainResource.foreach { path =>
+      classpathEntries.add(s"$path${File.separator}*")
+      val trinoDeps = Paths.get(path).getParent
+        .resolve(s"scala-$SCALA_COMPILE_VERSION")
+        .resolve("jars")
```
Comment (hddong) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilder.scala:75]: Related to [build/dist](https://github.com/apache/incubator-kyuubi/pull/2336/files#diff-b65bcfaa42cfe8af6eb1dba6b26df9ef944859901f355ec553592521fd8ade0d), seems path `scala-$SCALA_COMPILE_VERSION/jars` not exists.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilder.scala [hunk 3]
```diff
@@ -17,101 +17,71 @@
...
+    val classpathEntries = new LinkedHashSet[String]
+    // trino engine runtime jar
+    mainResource.foreach(classpathEntries.add)
 
-  override protected def childProcEnv: Map[String, String] = conf.getEnvs +
-    ("TRINO_ENGINE_JAR" -> mainResource.get) +
-    ("TRINO_ENGINE_DYNAMIC_ARGS" ->
-      trinoConf.map { case (k, v) => s"-D$k=$v" }.mkString(" "))
+    mainResource.foreach { path =>
+      classpathEntries.add(s"$path${File.separator}*")
```
Comment (hddong) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilder.scala:72]: This classpath seems like `${KYUUBI_HOME}/externals/engines/trino/kyuubi-trino-engine_2.12-1.6.0-SNAPSHOT.jar/*`, missing `getParent`?

File: kyuubi-server/src/test/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilderSuite.scala [hunk 4]
```diff
@@ -18,22 +18,25 @@
...
-import org.apache.kyuubi.config.KyuubiConf.ENGINE_TRINO_CONNECTION_URL
+import org.apache.kyuubi.config.KyuubiConf._
 
 class TrinoProcessBuilderSuite extends KyuubiFunSuite {
-  private def conf = KyuubiConf().set("kyuubi.on", "off")
 
   test("trino process builder") {
+    val conf = KyuubiConf()
+      .set(ENGINE_TRINO_CONNECTION_URL, "dummy_url")
+      .set(ENGINE_TRINO_CONNECTION_CATALOG, "dumm_catalog")
```
Comment (hddong) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilderSuite.scala:29]: nit: `dummy_catalog `

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilder.scala (hunk 3 above)
Comment (yaooqinn) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilder.scala:72]: nice catch!

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilder.scala (hunk 2 above)
Comment (yaooqinn) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilder.scala:75]: this seems not related to this pr

File: build/dist (hunk 1 above)
Comment (yaooqinn) [reply build/dist:257]: I don't agree, it make huge gaps between engines

Comment (yaooqinn) [reply build/dist:257]: what's more, most of the dependencies here
- duplicate with kyuubi servers, making our release tar oversize
- uncurated, we don't carefully manage the transitive jars, the behavior just inherits from the parent's maven-dependency-jar, which is dangerous for making an apache-approved release.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilder.scala (hunk 2 above)
Comment (yaooqinn) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilder.scala:75]: sorry, this is just for test

File: build/dist (hunk 1 above)
Comment (hddong) [reply build/dist:257]: Get it.

**Comment by hddong** [comment]
Thanks, LGTM

**Review by hddong — APPROVED** [review]

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilder.scala (hunk 2 above)
Comment (yaooqinn) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/trino/TrinoProcessBuilder.scala]: I made some changes here to make it clear. before I just added the prod/dev jars blindly. PTAL

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**